### PR TITLE
Deal with lava cans wasting lava

### LINKED ISSUE #531: Lava cans can waste lava
If you have use a full or partially full lava can to place lava, when the pointed node is a lava source node, the lava will disappear from the lava can but will not be placed into the world

NOTE: This might also apply to the water cans, it probably does, however, I haven't tested that, and that isn't as important, as you can make infinite water sources

### PR DESCRIPTION
**Pull request by BobFred7**
Should fix #531 
There are a bunch of unrelated commits here because I made the foolish mistake of committing to the master branch of my fork when dealing with a separate issue, but I think my reverts prevent that from causing any actual problems

### CODE REVIEW

File: technic/tools/cans.lua [hunk 1]
```diff
@@ -52,7 +52,7 @@ function technic.register_can(d)
 			if def.on_rightclick and user and not user:get_player_control().sneak then
 				return def.on_rightclick(pos, minetest.get_node(pos), user, itemstack, pointed_thing)
 			end
-			if not def.buildable_to then
+			if not def.buildable_to or minetest.get_node(pos).name == data.liquid_source_name then
 				pos = pointed_thing.above
 				def = minetest.registered_nodes[minetest.get_node(pos).name] or {}
 				if not def.buildable_to then return end
```
Comment (DS-Minetest) [inline technic/tools/cans.lua]: You should probably add the check here too.
Otherwise it might be possible to waste lava when swimming in lava.

Comment (BobFred7) [reply technic/tools/cans.lua]: You mean on line 58?

Comment (DS-Minetest) [reply technic/tools/cans.lua]: Yes, line 58 is where I made the code comment.
Just do the same as on line 55.

**Comment by BobFred7** [comment]
Been over a week and a half... any problems with this? is it going to be merged?

**Pull request merged**